src/qiskit_iqm/fake_backends/chip_sample.py

```diff
@@ -0,0 +1,155 @@
+            one_qubit_gate_depolarization_rates (dict[str, dict[int, float]]): for each one-qubit gate, a depolarization rate is specified for each qubit.
+            two_qubit_gate_depolarization_rates (dict[str, dict[tuple[int, int], float]]): for each two-qubit gate, a depolarization rate is specified for each coupling.
+            one_qubit_gate_durations (dict[str, float]): for each one-qubit gate, a duration is specified.
+            two_qubit_gate_durations (dict[str, float]): for each two-qubit gate, a duration is specified.
+            id_ (Union[str, None], optional): the identifier of the chip sample. Defaults to None.
+
+        Example:
+            .. code-block::
+
+                IQMChipSample(quantum_architecture=ThreeQubitExample(),
+                            t1s={0: 10000.0, 1: 12000.0, 2: 14000.0},
+                            t2s={0: 10000.0, 1: 12000.0, 3: 13000.0},
+                            one_qubit_gate_fidelities={"r": {0: 0.999,
+                                                            1: 0.996,
+                                                            2: 0.998}},
+                            two_qubit_gate_fidelities={"cz": {(0, 1): 0.995,
+                                                            (1, 2): 0.997}},
+                            one_qubit_gate_depolarization_rates={"r": {0: 0.0005,
+                                                                        1: 0.0004,
+                                                                        2: 0.0010}},
+                            two_qubit_gate_depolarization_rates={"cz": {(0, 1): 0.08,
+                                                                        (1, 2): 0.03}},
+                            one_qubit_gate_durations={"r": 50.},
+                            two_qubit_gate_durations={"cz": 100.},
+                            id_="threequbit-example_sample")
+        """
+        self.quantum_architecture = quantum_architecture
+        self.no_qubits = len(t1s)
+        self.t1s = t1s
+        self.t2s = t2s
+        self.one_qubit_gate_fidelities = one_qubit_gate_fidelities
+        self.two_qubit_gate_fidelities = two_qubit_gate_fidelities
+        self.gate_fidelities = self.one_qubit_gate_fidelities | self.two_qubit_gate_fidelities
+        self.one_qubit_gate_depolarization_rates = one_qubit_gate_depolarization_rates
+        self.two_qubit_gate_depolarization_rates = two_qubit_gate_depolarization_rates
+        self.gate_depolarization_rates = (
+            self.one_qubit_gate_depolarization_rates | self.two_qubit_gate_depolarization_rates
+        )
+        self.one_qubit_gate_durations = one_qubit_gate_durations
+        self.two_qubit_gate_durations = two_qubit_gate_durations
+        self.gate_durations = self.one_qubit_gate_durations | self.two_qubit_gate_durations
+        self.id_ = id_
```

Comment:
Have you considered using a Dataclass to reduce the boilerplate code here?

Reply:
uses dataclass now